### ISSUE TITLE
FEEDBACK

### ISSUE THREAD

**Issue by frksteenhoff** [frksteenhoff/ClimApp#82]
## User feedback
Feedback from ClimApp testers (1st beta version). All feedback will be listed here, the rate with which we implement the changes (and whether we are going to implement the changes) will be decided by the developer team.

Please add comments if you have any.

### Experienced/expert users
* ⛔️ not acted on
* ✔️ added to backlog

| Content screen | Comment | Contributor | Status  |
|--|--|--|--|
| **Settings**        |  |  |   |
|                            | Weather report not consistent, Chuansi and Jakob get different weather reports sitting in different rooms (in the same building), this might possibly be their device's implementation of geo ocation accuracy?  | Jakob Petersson, LU |  ⛔️  |
|                            | Set the value in settings scroll bar to current value to avoid having to scroll a lot  | Jakob Petersson, LU  | ✔️  |
|                            | Add more space between buttons in settings panel - currently very easy to misclick   |  Jakob Petersson, LU |  ✔️  |
|                            | Adjust `age`, `height`, `weight` values to match range of most narrow model. **Do they also work for children?** | Chuansi, LU |  ⛔️ |
|                            | Make it more intuitive that settings are clickable  | Chuansi, LU | ✔️  |
|                            | Make sure settings is saved correctly when entered by user | Chuansi, LU | ✔️   |
|                            |  |  |   |
| **Dashboard**   |  |  |   |
|                            | Make the current weather more clear/prominent on dashboard | Jade | ✔️  |
|                            | Add description of gauge and values/scales                                | Jade | ✔️  |
| **Onboarding** |  |  |   |
|                            | Send toast with "using default user settings" if none are input  | Chuansi, LU | ✔️ |
|                            | Under Welcome! “In order for the app to personalize the feedback provided to you …… ”. the feedback here can be confused with the feedback provided by users to ClimApp and/or to ClimApp data server. I suggest changing “feedback” to “thermal stress warnings and advice”.   | Chuansi, LU |  ⛔️ |
|                            | I suggest giving some indications (e.g. will not get personal advice) if no input personal info input. | Chuansi, LU | ⛔️  |
| **Other**           |  |  |   |
|                            | Add back button instead of having to click Dashboard/Settings  | Moniek, NL | ✔️   |
|                            | “Need location to get local weather reports”. To make it clearer why location is needed, I suggest adding “this app will use local weather report”. | Chuansi, LU |  ⛔️ |
|                            | Change logo (need the one to change to)  | Chuansi, LU | ⛔️  |
|                            |  |  |   |

### Inexperienced users

| Content screen | Comment | Contributor | Status  |
|--|--|--|--|
| **Settings**        |  |  |   |
|                            | *None*  |  |   |
| **Dashboard**   |  |  |   |
|                            | Unable to go back to current weather report can only choose the one 3 hours ahead if you have checked out a different time slot than now, |  | ⛔️  |
|                            | The vertical scroll on the dashboard is not so intuitive to figure out as there is no indication that scrolling is possible |  | ✔️ |
|                            | Understanding the values (highligthed below) for an entry-level user is hard. No indication of what it means/description of what the values indicate. Yes there is a gauge and the current value is described, but the scale is not defined properly (a possibility to explore the different values/an explanation for first time users would be needed). <img src="https://user-images.githubusercontent.com/6978512/55383470-87635e80-5528-11e9-9e48-225e02f5e597.png" alt="image" width="300"/> |  |  ⛔️  |
|                            | The `x% colder/warmer than yesterday` could possibly be changed into/be accompanied by a textual description like "it is very much colder than yesterday) The percentage make it more accurate but also more scientific which might not be the best solution for the end user.  |  | ⛔️   |
|                            | The showed temperature in the app varied a lot with the actual one at my living city, Lund. It’s been like this since I started using the app. **It is difficult to rely on this. I do not want to open another app in order to confirm current temperature while I am using my preferred app** for example, Climapp. | Amitava | ⛔️   |
|                            | It is not understandable at least for me by that is appearing ‘next 3 hours, next 6 hours…. Something in green color what is that? The time shows just below green colored numbers are different than next 3 hours. I am understood it is 3 hour after the current time. For example I’m using the app at 8 am and next 3 hours for me is 11 am but it shows 2 pm. It think that it not clear for most of us. | Amitava | ⛔️ |    
| **Onboarding**  |  |  |  |
|                            | On first sign in there should only be one possibility to choose Dashboard and Settings (not both the navigation bar and the buttons on the content panel -- seems like a bug that it is possible to "bypass the system" and makes for duplicate information which might seem confusing and redundant to the user. From this: <img src="https://user-images.githubusercontent.com/6978512/55727573-9196c880-5a12-11e9-87aa-563c52921db3.png" alt="image" width="200"/> to this: <img src="https://user-images.githubusercontent.com/6978512/55727593-9d828a80-5a12-11e9-96d0-aa65a7b105f6.png" alt="image" width="200"/> |  |  ✔️  |
|                            |  |  |   |
| **Other**            |  |  |   |
|                            | Consider placing the navigation bar at the bottom of the screen for easier access and navigation -- especially for long smart phones.  |  | ⛔️   |

## Suggestions to new functionality 
| Content screen | Comment | Contributor | Status  |
|--|--|--|--|
| **Settings**        |  |  |   |
|                            |  *None* |  |   |
| **Dashboard**   |  |  |   |
|                            | Look ahead several days | Moniek, NL  | ⛔️   |
|                            | Change location manually | Moniek, NL | ⛔️  |
|                            | Add/convert wind speed to Beaufort | Moniek, NL | ⛔️  |
| **Onboarding**  |  |  |   |
|                            | *None* |  |   |
| **Other**            |  |  |   |
|                            | (Possibly on new screen) Ability to input GPS coordinates for upcoming event /workout to get suggestions to clothing and hydration | Amitava  | ⛔️ |

**Comment by Chuansi-LU** [comment]
Settings related to Children and elderly: 
current thermal indices and models (WBGT, PHS, PMV and IREQ) are not designed for Children and the elderly. But the models can be adjusted to fit Children and the elderly. In the ClimApp Sciencetific Documents, we have included them as users. The following tasks and deliverables are copied from the document:
1.	Task T2.2., stakeholder VGGM in The Netherlands (focus in school children and elderly), 
2.	Task T2.4, school children; VGGM-study led by VU
3.	Deliverable: D 2.4: report of incorporated module for school children and the elderly.

**Comment by Chuansi-LU** [comment]
Here is the updated logo (I try to attach it here, not sure if it works, if does not work, I will send by email.)

**Comment by Chuansi-LU** [comment]
the app is working after my phone dropped to floor this morning during coffee break :). More comments below:

**Comment by Chuansi-LU** [comment]
1.	The gauge showed a values of “-1.2”, next 3 hours “-1.0”, etc. This value is somewhat confusing with the temperature (currently 9 °C). For a short moment, I was thinking “was the temperature outside -1.2 °C”? Suggest: add “cold stress level” after “-1.2, -1.0, etc.”, otherwise it can be easily confused with temperature.
2.	The warning message “-25% colder than yesterday 11:00”: this can be a kind of warning, although I am not sure what advice (actions) I should take if it is -25% or -45% colder or +45% warmer than yesterday? It gives only relative risk warning, but difficult to provide advice (adaptation strategies), the latter should also be an important integrated part of ClimApp.
3.	“Dress in layers to combat the cold stress”, which is based on “-1.2 cold stress”. I wonder if cold stress level is “-1.0”, “-1.8”, “2.2”, “-2.5” or “-3.7”, how the user dress should dress? Will the advice also be “Dress in layers”?
4.	Part of the text, e.g. activity level is outside the screen. The gauge and temperature sizes can be reduced in order to show all texts below the screen.
5.	“Details” for advanced users are very interesting. Change “Body temperature” to “Body heat balance” because “body temperature” in thermal physiology is a temperature weighed between skin and core temperature.
6.	On Details screen, some texts are outside the screen (need to scroll down to see more?)
7.	I think “Dashboard” and “Settings” do not need to be on the top of all screen pages all the time. I will attach or send a suggested FlowChart, where at the upper left corner of the screen, user can click an icon “---” and go to Settings, 
8.	The icon for “Rest” under Activity Level looks like the person is doing exercise (commented and suggested before by Stephen in his deliverable report). Suggest changing to an icon – sitting on sofa.
9.	There will be more comments later. According to the weather today, the prototype seems to have used IREQ model and WindChill temperature behind. Kalev is an experience researcher with IREQ model and cold environments. I suggest including him (kalev.kuklane@design.lth.se) in this internal test.

**Comment by Chuansi-LU** [comment]
ClimApp interface design flowchart suggestions, a
[climapp interface design flowchart_XG.pptx](https://github.com/frksteenhoff/ClimApp/files/3055091/climapp.interface.design.flowchart_XG.pptx)
ttached.

**Comment by BorisKingma** [comment]
nice! good to add to the troubleshoot faq! :-)

Op ma 8 apr. 2019 17:26 schreef Chuansi-LU <notifications@github.com>:

**Comment by BorisKingma** [comment]
thanks Chuansi for the detailed and valuable feedback. we list all feedback
and can report all findings soon

regards
boris

Op ma 8 apr. 2019 17:26 schreef Chuansi-LU <notifications@github.com>:

**Comment by Chuansi-LU** [comment]
I have just tested again. When clicked Settings, then my personal data disappeared (replaced by default data). Suggest: user's personal setting should be remained/saved there, unless the user clicks "Re-set personal settings". It would be nice if the user can click a button "My settings" like "My wardrobe", then my personal data/input will appear again (dob't need to input again).

**Comment by Chuansi-LU** [comment]
Another test just now, Tair=2 C, Cold stress level=-2.1, Tip of the moment: "Dress in layers to combact the cold stress". 
1. How much clothing insulation associated with "-2.1 cold stress" is not indicated somewhere. Is the default clothing insulation 1.0 clo used for the cold stress calculation?
2. In "Details", it would be nice if the user can further see "layers" and garments in next screen once the clothing icon is clicked.
3. A minor comment: change "combat" to "prevent".

**Comment by Chuansi-LU** [comment]
More comments after having tested the app for several consecutive Days when the weather is still Cold/cool. 
1. Wind speed seems to be fixed at 1 m/s all the time. for example, in Lund when the wind speed is about 6 m/s, but the app sitll shows 1 m/s. This may be the problem of OpenWeatherMap?
2. I have noticed that many time it suggests wearing windstopper even when wind speed low (1m/s) and WindChill temperature is not very Cold e.g. 4 C. Wind stopper clothing suggestion can be modified based on wind speed, e.g. >3 or 5m/s and WindChill temperature is Cold enough, e.g. <0 C.
3. The weather app (fromThe Weather Channel) seems to have weather data updated every hour? I noticed once at about 8:20 am, Weather app showed 3 C, but ClimApp showed 8 C (for 11:00 am), a big temperature difference (5 C).
4. The combination of low temperature, e.g. 8 C and Intense Activity, will lead to WBGT and PHS modules (on Details screen). I think WBGT will never reach threshold/reference value when temperature is <10 C. PHS may not be valid either (work time limitation and hydration suggesions) when temperature is < 10 C. For majority of population, this is still cold/cool environment, we probably cannot apply "heat module".
5. On details screen, sometimes the exposure time limit is a minus value, e.g. -83, -280866 minutes.
6. On details screen, "Radiation in full sunlight" e.g. 699 W/m2. How is this value converted to "mean radiant temperature" as input for IREQ? The outputs of IREQ after a few tests may suggest the mean radiant temperature inpuy is a bit too high. 
7. On Dashboard, we could try to keep only one activity Icon (e.g. medium) intead of 5 icons, and add one clothing Icon (e.g. business suit) on the same row, then add possiblity to go to next screen after clicking the activity or clothing icon to change activity and clothing level. As it is now, user is not very clear which clothing level (insulation) is used (default insulation input).
8. on Disclaimer screen, change "European Research Association ..." to "European Research Area..."

**Comment by BorisKingma** [comment]
Thanks Chuansi,

I have addressed your comments/questions below.

Kind regards,
Boris

   1. Wind speed seems to be fixed at 1 m/s all the time. for example, in
   Lund when the wind speed is about 6 m/s, but the app sitll shows 1 m/s.
   This may be the problem of OpenWeatherMap?


Windspeed shown in weather services is standard at 10m altitude. Windspeed
in Climapp is currently calculated windspeed at 2m, obtained from weather
forecast. Similar to how the windchill index converts windspeed (at 10m) to
windspeed at person level.



   1. I have noticed that many time it suggests wearing windstopper even
   when wind speed low (1m/s) and WindChill temperature is not very Cold e.g.
   4 C. Wind stopper clothing suggestion can be modified based on wind speed,
   e.g. >3 or 5m/s and WindChill temperature is Cold enough, e.g. <0 C.

Windstopper criterium is currently whenever Tair - Twindchill > 2.5C , as
far as I know, any criterium we use will be arbitrary because no study or
guideline exists. I do agree it should be different than currently in app.


   1. The weather app (fromThe Weather Channel) seems to have weather data
   updated every hour? I noticed once at about 8:20 am, Weather app showed 3
   C, but ClimApp showed 8 C (for 11:00 am), a big temperature difference (5
   C).


the weather forecast is for 11am, what was the temperature at 11am?

obtaining the "current" situation from openweathermap is an open issue, not
difficult to solve, but not in this version.

Asking "current", or "forecast" are 2 different things for weather services.

Furthermore, openweathermap is by my knowledge the only service that allows
up to 60 calls per minute for free. This is important, because other
services may give 1h forecasts, but easily charge >100Euro per month. This
is not sustainable for when ClimApp ends.

The climapp implementation stores the weather data, so that if somebody in
the neighbourhood uses the app at the same time, no new call to
openweathermap has to be made.

Also climapp stores the weather data so that a reference to yesterday can
actually be made. There is no weather service that allows for historical
data retrieval for free, and they are very, very expensive (e.g. 10 euro
per retrieval!, and there is no distinghuisment between 1day or 1 year of
historical data). So our solution seems the most practical way forward.


   1. The combination of low temperature, e.g. 8 C and Intense Activity,
   will lead to WBGT and PHS modules (on Details screen). I think WBGT will
   never reach threshold/reference value when temperature is <10 C. PHS may
   not be valid either (work time limitation and hydration suggesions) when
   temperature is < 10 C. For majority of population, this is still cold/cool
   environment, we probably cannot apply "heat module".

We can use a rule based system for this, but what should be the rules? We
should also be careful with ruling out the module information, because it
is the rational simulation result, not the subjective interpretation.


   1. On details screen, sometimes the exposure time limit is a minus
   value, e.g. -83, -280866 minutes.

yes, this is something I look into. It is an issue in the ISO
implementation. If I find the conditions this happens, we should inform ISO
on these cases, as IREQ or PHS fail on these thermal environemnts.

   1. On details screen, "Radiation in full sunlight" e.g. 699 W/m2. How is
   this value converted to "mean radiant temperature" as input for IREQ? The
   outputs of IREQ after a few tests may suggest the mean radiant temperature
   inpuy is a bit too high.

Tglobe is used for mean radiant temperature. But this is without taking
into account cloud cover, and also does not include the effect of wind.


   1. On Dashboard, we could try to keep only one activity Icon (e.g.
   medium) intead of 5 icons, and add one clothing Icon (e.g. business suit)
   on the same row, then add possiblity to go to next screen after clicking
   the activity or clothing icon to change activity and clothing level. As it
   is now, user is not very clear which clothing level (insulation) is used
   (default insulation input).

I agree on an iteration, yet I see within the climapp community we cannot
reach agreement on what a user should be able to input.


   1. on Disclaimer screen, change "European Research Association ..." to
   "European Research Area..."

done!

**Comment by Chuansi-LU** [comment]
Thank you Boris for the quick fix and feedback. A few responses below:

Best regards,
Chuansi

**Comment by frksteenhoff** [comment]
**Need to be added to list**, feedback from Amitava
Is nice to see that I don’t need to set repeatedly my subjective info. Great!

1. The showed temperature in the app varied a lot with the actual one at my living city, Lund. It’s been like this since I started using the app. It is difficult to rely on this. I do not want to open another app in order to confirm current temperature while I am using my preferred app for example, Climapp.

2. It is not understandable at least for me by that is appearing ‘next 3 hours, next 6 hours…. Something in green color what is that? The time shows just below green colored numbers are different than next 3 hours. I am understood it is 3 hour after the current time. For example I’m using the app at 8 am and next 3 hours for me is 11 am but it shows 2 pm. It think that it not clear for most of us.

 

However I opened the app with an expectation to get info about how the weather will be in next few hours when I wish to be outside for my physical training including running, plyometric etc.

In this case, I need a summary for the weather detail, my clothing types, no of layers (suggestion), hydration and so on. The suggestions can based on the activity, duration and weather.

 

Moreover, the tip is showing under each intensity of work does not match with activity levels and weather but the appeared ones are funny.
